### DISCUSSION TITLE
How to use Sqlite instead Posgresql

### DISCUSSION THREAD

**Discussion by auh83** [fastapi/full-stack-fastapi-template#1132] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
@computed_field  # type: ignore[misc]
    @property
    def SQLALCHEMY_DATABASE_URI(self) -> PostgresDsn:
        return MultiHostUrl.build(
            scheme="postgresql+psycopg",
            username=self.POSTGRES_USER,
            password=self.POSTGRES_PASSWORD,
            host=self.POSTGRES_SERVER,
            port=self.POSTGRES_PORT,
            path=self.POSTGRES_DB,
        )
```

### Description

I want to use sqlite instead Pstgresql, can you help me to replace it in the backend.

### Operating System

Windows

### Operating System Details

windows 10

### Python Version

python 3.11

**Comment by auh83** [comment]
@tiangolo I imagine there is a simple way to remplace Postgresql by sqlite !

**Reply by ubidev** [reply]
If the creator of FastAPI says so, then there's hope ;-)

**Comment by bwithai** [comment, marked as answer]
Hi @auh83 
Edit [SQLALCHEMY_DATABASE_URI](https://github.com/fastapi/full-stack-fastapi-template/blob/6453dea7ffc9cf6966b18e28b003b671e9592a9b/backend/app/core/config.py#L62C5-L70C10) with this code
```python
SQLITE_DB_FILE: str = ""
@computed_field  # type: ignore[prop-decorator]
@property
def SQLALCHEMY_DATABASE_URI(self) -> str | PostgresDsn:
        if self.SQLITE_DB_FILE:
            return f"sqlite:///{self.SQLITE_DB_FILE}"
        else:
            return MultiHostUrl.build(
            scheme="postgresql+psycopg",
            username=self.POSTGRES_USER,
            password=self.POSTGRES_PASSWORD,
            host=self.POSTGRES_SERVER,
            port=self.POSTGRES_PORT,
            path=self.POSTGRES_DB,
        )
```

and in your .env file `SQLITE_DB_FILE=my_db.sqlite` --> `python backend\app\initial_data.py`
